tests/unit_tests/test_peps.py

```diff
@@ -0,0 +1,234 @@
+                   78, 77, 73, 50, 50, 56, 68, 98, 95, 112, 134, 145, 158,
+                   172, 213, 234, 222, 215, 216, 216, 206, 183, 135, 156, 110, 92,
+                   63, 60, 52, 29, 20, 16, 12, 5, 5, 5, 1, 2, 3,
+                   0, 2])
+    k0 = np.array([0., 0., 0., 0., 0., 0.,
+                   0., 0., 0.,
+                   0., 0., 0.,
+                   0., 0., 0.,
+                   0., 0., 0.,
+                   0., 0., 0.7205812007860187,
+                   0., 1.4411624015720375, 0.7205812007860187,
+                   2.882324803144075, 5.76464960628815, 5.76464960628815,
+                   12.249880413362318, 15.132205216506394, 20.176273622008523,
+                   27.382085629868712, 48.27894045266326, 47.558359251877235,
+                   68.45521407467177, 97.99904330689854, 108.0871801179028,
+                   135.46926574777152, 140.51333415327366, 184.4687874012208,
+                   171.49832578707245, 205.36564222401535, 244.27702706646033,
+                   214.01261663344755, 228.42424064916793, 232.02714665309804,
+                   205.36564222401535, 172.9394881886445, 191.67459940908097,
+                   162.1307701768542, 153.48379576742198, 110.96950492104689,
+                   103.04311171240067, 86.46974409432225, 60.528820866025576,
+                   43.234872047161126, 23.779179625938617, 24.499760826724636,
+                   17.29394881886445, 11.5292992125763, 5.76464960628815,
+                   5.044068405502131, 3.6029060039300935, 0.,
+                   2.882324803144075, 0., 0.,
+                   0.])
+    p = peps.fit_nnls(n0, k0, ascending=True, erase_zeros=True)
+    assert np.all(np.diff(p) >= 0)
+    assert np.linalg.norm(k0 - n0 * p, np.inf) < 40
+    p = peps.fit_nnls(n0, k0, ascending=True, erase_zeros=not True)
+    assert np.all(np.diff(p) >= 0)
+    assert np.linalg.norm(k0 - n0 * p, np.inf) < 40
+
+
+def test_peps_qvality():
+    scores, targets = get_target_decoy_data()
+    peps_values = peps.peps_from_scores_qvality(scores, targets)
+    assert np.all(peps_values >= 0)
+    assert np.all(peps_values <= 1)
+    assert np.all(np.diff(peps_values) * np.diff(scores) <= 0)
+
+
+def test_peps_kde_nnls():
+    np.random.seed(1253)  # this produced an error with failing iterations in nnls
+    scores, targets = get_target_decoy_data()
+    peps_values = peps.peps_from_scores_kde_nnls(scores, targets)
+    assert np.all(peps_values >= 0)
+    assert np.all(peps_values <= 1)
+    assert np.all(np.diff(peps_values) * np.diff(scores) <= 0)
+
+    np.random.seed(1245) # this produced an assertion error due to peps over 1.0
+    scores, targets = get_target_decoy_data()
+    peps_values = peps.peps_from_scores_kde_nnls(scores, targets)
+    assert np.all(peps_values >= 0)
+    assert np.all(peps_values <= 1)
+    assert np.all(np.diff(peps_values) * np.diff(scores) <= 0)
+
+
+def test_peps_hist_nnls():
```

Comment:
This can probably use `@pytest.mark.parametrize`
Would yo mind providing more context on the errors? and exactly what you want to test here.